### DISCUSSION TITLE
Load components from disk at runtime

### DISCUSSION THREAD

**Discussion by euroUK** [stride3d/stride#2233] [Newcomers] [answered]
Hi!

Thank you for your awesome job. Can you please help me with loading components. I'm trying to load *.sduipage at runtime via code only aproach, and it looks like  Content.Load<UIPage>("UI/Main") loads only compiled components.

**Comment by FloccinaucinihilifilipicatesLongNames** [comment, marked as answer]
Hello, sorry for the late response, I just stumbled upon this question.

If you're looking to load a UIPage at runtime, the way to go about it is:
- Create a UIComponent (preferably defined globally): `UIComponent MyUI = new();`
- Add the page to the component: `MyUI.Page = Content.Load<UIPage>("UI/Main");`
- Add the UIComponent to the entity that has the script attached (or any entity in the scene, really): `Entity.Components.Add(MyUI);`
- Make sure the UIPage is included in the build as a root asset in the editor.

If the issue is more specific or you need more help with this, feel free to let me know with additional info and I'll do my best to look into it.
I hope this helps!
